transport_nantes/mailing_list/views.py

```diff
@@ -252,9 +254,78 @@ class MailingListListView(ListView):
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
-        lists = self.get_queryset()
-        context['mailing_lists'] = [(list, user_subscribe_count(list))
-                                    for list in lists if not list.is_petition]
-        context['petitions_lists'] = [(list, subscriber_count(list))
-                                      for list in lists if list.is_petition]
+        mailing_lists = self.get_queryset()
+        context['mailing_lists'] = [(mailing_list, user_subscribe_count(
+            mailing_list)) for mailing_list in mailing_lists
+            if not mailing_list.is_petition]
+        context['petitions_lists'] = [(mailing_list, subscriber_count(
+            mailing_list)) for mailing_list in mailing_lists
+            if mailing_list.is_petition]
         return context
+
+
+class UserStatusView(LoginRequiredMixin, ListView):
+    model = MailingList
+    template_name = "mailing_list/mailing_list_user_status.html"
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['mailing_lists'] = list()
+        context['petitions_lists'] = list()
+        base_lists = MailingList.objects.filter(
+            list_active=True).order_by(
+                'is_petition', 'mailing_list_name')
+        for mailing_list in base_lists:
+            state = user_current_state(
+                self.request.user, mailing_list).event_type
+            if not mailing_list.is_petition:
+                context['mailing_lists'].append((mailing_list, state,))
+            else:
+                if state != "sub":
+                    context['petitions_lists'].append((mailing_list, state,))
+        context['user_id'] = self.request.user.id
+        return context
+
+
+class MailingListSubscribeFromStatus(LoginRequiredMixin, FormView):
```

Comment:
I feel like subscribre and unsuscribe could be in the same view, where depending on the current status, you either invoke `subscribe_user_to_list` or `unsubscribe_user_from_list`.
The two views are extremely similar : They both check, quite unnecessarily (?), the user id of the form against the one of the request, and then get user and ML objects to call the approriate function. 
Why wouldn't you use `user_current_state` with user and ML as args to call one of the two function ? This way you could have a unique "MailingListToggleSubscription" instead of two separate views.